### ISSUE TITLE
The 'sysroot.specs' feature causes GCC failure for the 'nweb' bundle on Ubuntu 

### ISSUE THREAD

**Issue by ivandavidov** [ivandavidov/minimal#80]
This is the official 'bug issue' for #67.

The command which causes the bug is this one:

```
ivan@ubuntu:~/minimal/src/minimal_overlay/bundles/nweb$ gcc -specs=/home/ivan/minimal/src/work/sysroot.specs -static-libgcc -Wl,-nostdlib -Os -s -fno-stack-protector -fomit-frame-pointer -U_FORTIFY_SOURCE -Wl,-nostdlib /home/ivan/minimal/src/minimal_overlay/bundles/nweb/nweb23.c -o nweb
/usr/bin/ld: cannot find /lib/libc.so.6
/usr/bin/ld: cannot find /lib/libc_nonshared.a
/usr/bin/ld: cannot find /lib/ld-linux-x86-64.so.2
collect2: error: ld returned 1 exit status
```

The contents of the ``sysroot.specs`` file is the following:

```
*link_libgcc:
-L/home/ivan/minimal/src/work/sysroot/lib -L/home/ivan/minimal/src/work/sysroot/lib64 -L/home/ivan/minimal/src/work/sysroot/usr/lib -L/home/ivan/minimal/src/work/sysroot/usr/lib64 -L/home/ivan/minimal/src/work/sysroot/usr/local/lib -L/home/ivan/minimal/src/work/sysroot/usr/local/lib64 -L/usr/lib/gcc/x86_64-linux-gnu/5
```

GCC is 5.4.0 on Ubuntu 16.04.
MLL kernel is 4.12.4, glibc is 2.26 and BusyBox is 1.27.1 (BTW it doesn't matter).

I'm not yet familiar with the ``sysroot.specs`` functionality (yep, my bad) and I'll be glad if anyone could assist me and figure out what the problem is and how I can fix it permanently.

**Comment by bauen1** [comment]
Can you confirm if each of the paths exists ?
/home/ivan/minimal/src/work/sysroot/lib
/home/ivan/minimal/src/work/sysroot/lib64
/home/ivan/minimal/src/work/sysroot/usr/lib
/home/ivan/minimal/src/work/sysroot/usr/lib64
/home/ivan/minimal/src/work/sysroot/usr/local/lib
/home/ivan/minimal/src/work/sysroot/usr/local/lib64
/usr/lib/gcc/x86_64-linux-gnu/5
The last one seems odd to me.
I should have noted that the sysroot.specs file needs to be regenerated if you move the projects location because it contains absolute paths.

**Comment by ivandavidov** [comment]
```
ivan@ubuntu:~$ 
ivan@ubuntu:~$ cd /home/ivan/minimal/src/work/sysroot/lib
ivan@ubuntu:~/minimal/src/work/sysroot/lib$ cd /home/ivan/minimal/src/work/sysroot/lib64
bash: cd: /home/ivan/minimal/src/work/sysroot/lib64: No such file or directory
ivan@ubuntu:~/minimal/src/work/sysroot/lib$ cd /home/ivan/minimal/src/work/sysroot/usr/lib
ivan@ubuntu:~/minimal/src/work/sysroot/usr/lib$ cd /home/ivan/minimal/src/work/sysroot/usr/lib64
bash: cd: /home/ivan/minimal/src/work/sysroot/usr/lib64: No such file or directory
ivan@ubuntu:~/minimal/src/work/sysroot/usr/lib$ cd /home/ivan/minimal/src/work/sysroot/usr/local/lib
bash: cd: /home/ivan/minimal/src/work/sysroot/usr/local/lib: No such file or directory
ivan@ubuntu:~/minimal/src/work/sysroot/usr/lib$ cd /home/ivan/minimal/src/work/sysroot/usr/local/lib64
bash: cd: /home/ivan/minimal/src/work/sysroot/usr/local/lib64: No such file or directory
ivan@ubuntu:~/minimal/src/work/sysroot/usr/lib$ cd /usr/lib/gcc/x86_64-linux-gnu/5
ivan@ubuntu:/usr/lib/gcc/x86_64-linux-gnu/5$ 
```

The ``lib64`` directory doesn't exist because glibc is compiled with ``--prefix= \`` and all shared objects are generated in the ``src/work/sysroot/lib`` folder.

The last folder is computed from the output of ``gcc -print-libgcc-file-name`` and in my case this gives ``/usr/lib/gcc/x86_64-linux-gnu/5/libgcc.a``.

**Comment by bauen1** [comment]
Are you sure that the sysroot (glibc, etc ...) have been compiled ?
There should not be any issue with the spec file itself.

**Comment by ivandavidov** [comment]
The sysroot is fine and I can produce MLL ISO image which also works fine. I have only tried the ``dhcp`` and ``nweb`` bundles so far. The ``dhcp`` bundle works fine (there is no compilation work there) and MLL gets IP via DHCP but the ``nweb`` bundle fails and the actual binary is not produced.

I have pushed my latest changes. Using the latest kernel and glibc shouldn't be the root cause of this issue. Nevertheless, I'll also give it a try with the older kernel and glibc.
